src/libstd/thread/local.rs

```diff
@@ -359,13 +456,25 @@ impl<T: 'static> LocalKey<T> {
     pub fn try_with<F, R>(&'static self, f: F) -> Result<R, AccessError>
                       where F: FnOnce(&T) -> R {
         unsafe {
-            let slot = (self.inner)().ok_or(AccessError {
-                _private: (),
-            })?;
-            Ok(f(match *slot.get() {
-                Some(ref inner) => inner,
-                None => self.init(slot),
-            }))
+            let slot = (self.get)();
+            if let &LocalKeyValue::Valid(ref inner) = &*slot.get() {
+                // Do this in a separate if branch (rather than just part of the
+                // match statement in the else block) to increase the performance
+                // of the fast path.
```

Comment:
Does this measurably help the codegen?

I'd naively expect not, butI'm curious if you have benchmarks.

Reply:
So I'm actually not sure. I tried benchmarking it, but I'm not confident that my `xargo` command is testing what I think it is because I ran it with a version of the code that had a compiler error, and it still ran properly, so I'm not sure what artifact `xargo` is picking up to link as libstd. Here's what I'm running:

`RUST_BACKTRACE=1 RUSTFLAGS='-C opt-level=3' XARGO_RUST_SRC=/path/to/rust/src/ xargo run --release`

Given this command, it looks like there's not much of a difference, but I'm skeptical for the reasons mentioned. If you want to give it a try yourself (since I assume you're better at this whole build pipeline thing), it would be give me more confidence.